validator/src/journal/block_manager.rs

```diff
@@ -583,7 +581,7 @@ impl BranchIterator {
         };
         BranchIterator {
             state,
-            initial_block_id: first_block_id.clone(),
+            initial_block_id: next_block_id.clone(),
```

Comment:
Is this a typo, or should it be in a different commit?

Reply:
Yeah, that looks like it should be in the previous commit, as should the changes to drop